tests/test_Config.cpp

```diff
@@ -269,36 +269,40 @@ BOOST_AUTO_TEST_CASE(test_TempFile) {
     BOOST_CHECK(!rwfs::exists(path));
 }
 
-BOOST_AUTO_TEST_CASE(test_config_initial) {
+BOOST_AUTO_TEST_CASE(test_configparser_initial) {
     // Test an initial config
-    GameConfig cfg;
-    BOOST_CHECK(!cfg.isValid());
+    [[maybe_unused]] RWConfigParser cfgParser;
 }
 
-BOOST_AUTO_TEST_CASE(test_config_valid) {
+BOOST_AUTO_TEST_CASE(test_configparser_valid) {
     // Test reading a valid configuration file
     auto cfg = getValidConfig();
 
     TempFile tempFile;
     tempFile.append(cfg);
 
-    GameConfig config;
-    config.loadFile(tempFile.path());
+    RWConfigParser cfgParser;
+    RWConfigLayer cfgLayer;
+    RWConfigParser::ParseResult parseResult;
+    std::tie(cfgLayer, parseResult) = cfgParser.loadFile(tempFile.path());
```

Comment:
You can avoid using `std::tie` using structure bindings.

Reply:
That would have saved me a lot of typing :D